go/vt/vtgate/executor.go

```diff
@@ -766,10 +793,20 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, bindVars map[string
 	if err != nil {
 		return nil, err
 	}
-	e.plans.Set(normkey, plan)
+	if !noPlanCache {
+		e.plans.Set(normkey, plan)
+	}
 	return plan, nil
 }
 
+// getNoCacheQueryPlanFromSession extracts NoCacheQueryPlan from session
```

Comment:
This is a mouthful and the "FromSession" is implicit in the function signature.

Also in light of the above comment, I'd change this to be `func shouldCacheQueryPlan(session *vtgatepb.Session) bool`